# Exploring potential new hotel venues in Brussels

### 1. Introduction

My report is for those who are planning to start a new hotel in the city of Brussels. I will provide a suggestion on what would be the best venue to start a new hotel in this densely populated and highly visited city.

Brussels, officially the Brussels-Capital Region, is a region of Belgium comprising 19 municipalities, including the City of Brussels, which is the capital of Belgium. The Brussels-Capital Region is located in the central portion of the country and is a part of both the French Community of Belgium and the Flemish Community, but is separate from the Flemish Region (in which it forms an enclave) and the Walloon Region. Brussels is the most densely populated and the richest region in Belgium in terms of GDP per capita. 

Unassuming Brussels is the capital of Belgium, Flanders and Europe. Medieval Grand-Place, is indeed grand, with many 17th-century buildings and daily flower markets. Reopened in 2006, the Atomium, Brussels' Eiffel Tower, provides great views, inside and out. Architecture fans should visit Musee Horta, home of Belgian master architect Victor Horta. St. Gery's clubs and bars are packed year-round. Seafood eateries abound in Ste. Catherine. Walk, rather than get snarled up in traffic, in the narrow streets.

### 2. Business Problem

In my report, I will focus on the issue of where to open a new hotel in a city like Brussels, once one has decided to go ahead. Let’s imagine Marriott Hotels are willing to open a new luxury hotel, the first and foremost important decision will be the location for its new hotel.
1. On what basis can they decide the new hotel's location?
2. While selecting the place there are key points to consider like they need to check where the most well-visited venues of the city are?
3. If incase there are already other luxury hotels which have good ratings, will it be risky to open new one near these hotels?
4. Out of scope for this project: Rent and land values in the neighborhoods, budget for the interior decoration of the hotel, budget for opening different restaurants in the hotel etc.

### 3. Data Preperation

As we are creating a report for those who want to open a new luxury hotel in Brussels, the first requirement is to collect Brussels postal codes data with the name of respective neighborhoods. The second requirement would be collect data related to latitude and longitude values of the same neighbrohoods and merge the two datasets.

There are 19 municipalities in Brussels with different neighborhoods. We will explore each municipality and their respective neighborhoods to check which neighborhood has the most visited venues and would be perfect to open a new hotel.
We will extract the data from below wikipedia page using Beautiful Soup.
https://en.wikipedia.org/wiki/List_of_municipalities_of_the_Brussels-Capital_Region

We will extract the data related to latitude and logitude values of Neighborhoods with a csv file(zipcode-belgium.csv) saved at local machine and then at the server. 

Now once we have the latitude and longitude data, let's use Foursquare Location to get the amount of most visited venues per Neighborhood, which will give us an idea of where the tourist are moving when visiting the city. This will already show us the best Neighborhoods to start a hotel. The details can be retrieved using search endpoint.
Link to the dataset is: https://developer.foursquare.com/docs/data

#### Use the BeautifulSoup package to transform data in the table on the Wikipedia page into pandas dataframe.
#### Importing libraries to get data in required format. 

In [44]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

In [45]:
url = requests.get('https://en.wikipedia.org/wiki/List_of_municipalities_of_the_Brussels-Capital_Region').text

In [46]:
soup = BeautifulSoup(url,'lxml')
print(soup.prettify())

<!DOCTYPE html>
<html class="client-nojs" dir="ltr" lang="en">
 <head>
  <meta charset="utf-8"/>
  <title>
   List of municipalities of the Brussels-Capital Region - Wikipedia
  </title>
  <script>
   document.documentElement.className="client-js";RLCONF={"wgBreakFrames":!1,"wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRequestId":"Xd4X5ApAICkAADuMvHsAAACI","wgCSPNonce":!1,"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":!1,"wgNamespaceNumber":0,"wgPageName":"List_of_municipalities_of_the_Brussels-Capital_Region","wgTitle":"List of municipalities of the Brussels-Capital Region","wgCurRevisionId":923747618,"wgRevisionId":923747618,"wgArticleId":261746,"wgIsArticle":!0,"wgIsRedirect":!1,"wgAction":"view","wg

#### We can see that the data required is available in table and belongs to class 'wikitable sortable'. So, let's extract only the table data

In [47]:
My_table = soup.find('table',{'class':'wikitable sortable'})
My_table

<table class="wikitable sortable">
<tbody><tr>
<th>
</th>
<th>French name
</th>
<th>Dutch name
</th>
<th class="unsortable"><a href="/wiki/Flag" title="Flag">Flag</a>
</th>
<th class="unsortable"><a href="/wiki/Coat_of_arms" title="Coat of arms">CoA</a>
</th>
<th><a href="/wiki/Postal_code" title="Postal code">post<br/><small>code</small></a>
</th>
<th>Population<br/><small>(1/1/2017)</small>
</th>
<th>Area
</th>
<th><a href="/wiki/Population_density" title="Population density">Population density</a><br/><small>(km²)</small>
</th>
<th class="unsortable">Ref.
</th></tr>
<tr>
<td style="text-align:center;">1
</td>
<td style="text-align:left"><a href="/wiki/Anderlecht" title="Anderlecht">Anderlecht</a>
</td>
<td style="text-align:left"><a href="/wiki/Anderlecht" title="Anderlecht">Anderlecht</a>
</td>
<td style="text-align:center;"><a class="image" href="/wiki/File:Flag_of_Anderlecht.svg"><img alt="Flag of Anderlecht.svg" class="thumbborder" data-file-height="400" data-file-width="600" de

In [48]:
tableRows = [[td.get_text() for td in row.find_all('td')] for row in My_table.find_all('tr')[1:]]
print(tableRows)

[['1\n', 'Anderlecht\n', 'Anderlecht\n', '\n', '\n', '1070\n', '118,241\n', '17.717.7\xa0km2 (6.8\xa0sq\xa0mi)\n', '6,680\n'], ['2\n', 'Auderghem\n', 'Oudergem\n', '\n', '\n', '1160\n', '33,313\n', '09.09.0\xa0km2 (3.5\xa0sq\xa0mi)\n', '3,701\n'], ['3\n', 'Berchem-Sainte-Agathe\n', 'Sint-Agatha-Berchem\n', '\n', '\n', '1082\n', '24,701\n', '02.92.9\xa0km2 (1.1\xa0sq\xa0mi)\n', '8,518\n'], ['4\n', 'Bruxelles-Ville*\n', 'Stad Brussel*\n', '\n', '\n', '1000102011201130\n', '176,545\n', '32.632.6\xa0km2 (12.6\xa0sq\xa0mi)\n', '5,415\n'], ['5\n', 'Etterbeek\n', 'Etterbeek\n', '\n', '\n', '1040\n', '47,414\n', '03.13.1\xa0km2 (1.2\xa0sq\xa0mi)\n', '15,295\n'], ['6\n', 'Evere\n', 'Evere\n', '\n', '\n', '1140\n', '40,394\n', '05.05.0\xa0km2 (1.9\xa0sq\xa0mi)\n', '8,079\n'], ['7\n', 'Forest\n', 'Vorst\n', '\n', '\n', '1190\n', '55,746\n', '06.26.2\xa0km2 (2.4\xa0sq\xa0mi)\n', '8,991\n'], ['8\n', 'Ganshoren\n', 'Ganshoren\n', '\n', '\n', '1083\n', '24,596\n', '02.52.5\xa0km2 (1.0\xa0sq\xa0mi)\n'

#### Let's extract the data into pandas dataframe

In [49]:
tableHeaders = ["Index","Neighborhood","Dutch name","Flag","CoA","Postal code","Population","Area","Population density"]
df = pd.DataFrame(tableRows, columns=tableHeaders)
df.head(10)

Index             Neighborhood             Dutch name Flag CoA  \
0   1\n             Anderlecht\n           Anderlecht\n   \n  \n   
1   2\n              Auderghem\n             Oudergem\n   \n  \n   
2   3\n  Berchem-Sainte-Agathe\n  Sint-Agatha-Berchem\n   \n  \n   
3   4\n       Bruxelles-Ville*\n        Stad Brussel*\n   \n  \n   
4   5\n              Etterbeek\n            Etterbeek\n   \n  \n   
5   6\n                  Evere\n                Evere\n   \n  \n   
6   7\n                 Forest\n                Vorst\n   \n  \n   
7   8\n              Ganshoren\n            Ganshoren\n   \n  \n   
8   9\n                Ixelles\n               Elsene\n   \n  \n   
9  10\n                  Jette\n                Jette\n   \n  \n   

          Postal code Population                         Area  \
0              1070\n  118,241\n   17.717.7 km2 (6.8 sq mi)\n   
1              1160\n   33,313\n    09.09.0 km2 (3.5 sq mi)\n   
2              1082\n   24,701\n    02.92.9 km2 (1.1 sq mi)\n   
3  1000102011201130\n  176,545\n  32.632.6 km2 (12.6 sq mi)\n   
4              1040\n   47,414\n    03.13.1 km2 (1.2 sq mi)\n   
5              1140\n   40,394\n    05.05.0 km2 (1.9 sq mi)\n   
6              1190\n   55,746\n    06.26.2 km2 (2.4 sq mi)\n   
7              1083\n   24,596\n    02.52.5 km2 (1.0 sq mi)\n   
8              1050\n   86,244\n    06.36.3 km2 (2.4 sq mi)\n   
9              1090\n   51,933\n    05.05.0 km2 (1.9 sq mi)\n   

  Population density  
0            6,680\n  
1            3,701\n  
2            8,518\n  
3            5,415\n  
4           15,295\n  
5            8,079\n  
6            8,991\n  
7            9,838\n  
8           13,690\n  
9           10,387\n

#### Let's refine and transfrom data as required

In [50]:
df = df.replace('\n','', regex=True)
df.head()

Index           Neighborhood           Dutch name Flag CoA  \
0     1             Anderlecht           Anderlecht            
1     2              Auderghem             Oudergem            
2     3  Berchem-Sainte-Agathe  Sint-Agatha-Berchem            
3     4       Bruxelles-Ville*        Stad Brussel*            
4     5              Etterbeek            Etterbeek            

        Postal code Population                       Area Population density  
0              1070    118,241   17.717.7 km2 (6.8 sq mi)              6,680  
1              1160     33,313    09.09.0 km2 (3.5 sq mi)              3,701  
2              1082     24,701    02.92.9 km2 (1.1 sq mi)              8,518  
3  1000102011201130    176,545  32.632.6 km2 (12.6 sq mi)              5,415  
4              1040     47,414    03.13.1 km2 (1.2 sq mi)             15,295

#### Let's drop the columns that are not required

In [51]:
df.drop(['Index', 'Dutch name', 'Flag','CoA','Population','Area','Population density'], inplace = True, axis = 1)
df.head()

Neighborhood       Postal code
0             Anderlecht              1070
1              Auderghem              1160
2  Berchem-Sainte-Agathe              1082
3       Bruxelles-Ville*  1000102011201130
4              Etterbeek              1040

In [52]:
df['Postal code'] = df['Postal code'].replace('1000102011201130','1000')
df.head()

Neighborhood Postal code
0             Anderlecht        1070
1              Auderghem        1160
2  Berchem-Sainte-Agathe        1082
3       Bruxelles-Ville*        1000
4              Etterbeek        1040

In [53]:
df['Postal code']= df['Postal code'].astype(int)
df

Neighborhood  Postal code
0              Anderlecht         1070
1               Auderghem         1160
2   Berchem-Sainte-Agathe         1082
3        Bruxelles-Ville*         1000
4               Etterbeek         1040
5                   Evere         1140
6                  Forest         1190
7               Ganshoren         1083
8                 Ixelles         1050
9                   Jette         1090
10             Koekelberg         1081
11   Molenbeek-Saint-Jean         1080
12           Saint-Gilles         1060
13  Saint-Josse-ten-Noode         1210
14             Schaerbeek         1030
15                  Uccle         1180
16    Watermael-Boitsfort         1170
17   Woluwe-Saint-Lambert         1200
18    Woluwe-Saint-Pierre         1150

#### Now let's add the csv file containing latitude and logitude values to server and then impirt the data into pandas dataframe

In [54]:
# The code was removed by Watson Studio for sharing.

Postal code Neighborhood   Latitude  Longitude
0         1000    Bruxelles  50.846557   4.351697
1         1020       Laeken  50.883392   4.348713
2         1030   Schaerbeek  50.867604   4.373712
3         1040    Etterbeek  50.836851   4.389510
4         1050      Ixelles  50.822285   4.381571

#### Let's merge two dataframes into a single dataframe for further analysis

In [55]:
Brussel_df = pd.merge(df,df_data_0, on='Postal code')
Brussel_df

Neighborhood_x  Postal code         Neighborhood_y   Latitude  \
0              Anderlecht         1070             Anderlecht  50.838141   
1               Auderghem         1160              Auderghem  50.815657   
2   Berchem-Sainte-Agathe         1082  Berchem-Sainte-Agathe  50.863984   
3        Bruxelles-Ville*         1000              Bruxelles  50.846557   
4               Etterbeek         1040              Etterbeek  50.836851   
5                   Evere         1140                  Evere  50.870452   
6                  Forest         1190                 Forest  50.809143   
7               Ganshoren         1083              Ganshoren  50.871240   
8                 Ixelles         1050                Ixelles  50.822285   
9                   Jette         1090                  Jette  50.877763   
10             Koekelberg         1081             Koekelberg  50.862263   
11   Molenbeek-Saint-Jean         1080   Molenbeek-Saint-Jean  50.854355   
12           Saint-Gilles         1060           Saint-Gilles  50.826741   
13  Saint-Josse-ten-Noode         1210  Saint-Josse-Ten-Noode  50.853074   
14             Schaerbeek         1030             Schaerbeek  50.867604   
15                  Uccle         1180                  Uccle  50.801820   
16    Watermael-Boitsfort         1170    Watermael-Boitsfort  50.799394   
17   Woluwe-Saint-Lambert         1200   Woluwe-Saint-Lambert  50.846693   
18    Woluwe-Saint-Pierre         1150    Woluwe-Saint-Pierre  50.829243   

    Longitude  
0    4.312340  
1    4.433139  
2    4.292702  
3    4.351697  
4    4.389510  
5    4.402160  
6    4.317751  
7    4.317510  
8    4.381571  
9    4.326090  
10   4.325708  
11   4.322778  
12   4.345668  
13   4.372336  
14   4.373712  
15   4.337235  
16   4.415818  
17   4.428484  
18   4.443297

In [56]:
Brussel_df.drop(['Neighborhood_x'], inplace = True, axis = 1)
Brussel_df.head()

Postal code         Neighborhood_y   Latitude  Longitude
0         1070             Anderlecht  50.838141   4.312340
1         1160              Auderghem  50.815657   4.433139
2         1082  Berchem-Sainte-Agathe  50.863984   4.292702
3         1000              Bruxelles  50.846557   4.351697
4         1040              Etterbeek  50.836851   4.389510

In [57]:
Brussel_df.rename(columns={'Neighborhood_y':'Neighborhood'}, inplace=True)
Brussel_df.head()

Postal code           Neighborhood   Latitude  Longitude
0         1070             Anderlecht  50.838141   4.312340
1         1160              Auderghem  50.815657   4.433139
2         1082  Berchem-Sainte-Agathe  50.863984   4.292702
3         1000              Bruxelles  50.846557   4.351697
4         1040              Etterbeek  50.836851   4.389510

### Let's Explore and cluster the neighborhoods in Brussels

#### Use geopy library to set latitude and longitude values of Brussels

In [59]:
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes
import folium # map rendering library

print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Libraries imported.


In [60]:
address = 'Brussel, Belgium'

geolocator = Nominatim(user_agent="Brussel")
location = geolocator.geocode(address)
latitude_Brussel = location.latitude
longitude_Brussel = location.longitude
print('The geograpical coordinate of Brussel are {}, {}.'.format(latitude_Brussel, longitude_Brussel))

The geograpical coordinate of Brussel are 50.8436709, 4.36743669338796.


In [61]:
map_Brussel = folium.Map(location=[latitude_Brussel, longitude_Brussel], zoom_start=10)

# add markers to map
for lat, lng, Neighbourhood in zip(Brussel_df['Latitude'], Brussel_df['Longitude'], Brussel_df['Neighborhood']):
    label = '{}'.format(Neighbourhood)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_Brussel)  
    
map_Brussel

#### Define foursquare credentials and version

In [62]:
# The code was removed by Watson Studio for sharing.

#### Let's check all the neighborhoods in Brussels

In [63]:
radius = 500
LIMIT = 100

In [64]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [65]:
Brussel_venues = getNearbyVenues(names=Brussel_df['Neighborhood'],
                                   latitudes=Brussel_df['Latitude'],
                                   longitudes=Brussel_df['Longitude']
                                  )

Anderlecht
Auderghem
Berchem-Sainte-Agathe
Bruxelles
Etterbeek
Evere
Forest
Ganshoren
Ixelles
Jette
Koekelberg
Molenbeek-Saint-Jean
Saint-Gilles
Saint-Josse-Ten-Noode
Schaerbeek
Uccle
Watermael-Boitsfort
Woluwe-Saint-Lambert
Woluwe-Saint-Pierre


#### Let's check the size of resulting dataframe

In [66]:
print(Brussel_venues.shape)
Brussel_venues.head()

(605, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0   Anderlecht              50.838141                 4.31234   
1   Anderlecht              50.838141                 4.31234   
2   Anderlecht              50.838141                 4.31234   
3   Anderlecht              50.838141                 4.31234   
4   Anderlecht              50.838141                 4.31234   

                                Venue  Venue Latitude  Venue Longitude  \
0                        Friture René       50.835846         4.311632   
1                        Snack Mirvan       50.835176         4.308543   
2                    Le Chapeau Blanc       50.835034         4.307790   
3                              Ulysse       50.838612         4.306860   
4  Bospark / Parc Forestier (Bospark)       50.840087         4.310731   

       Venue Category  
0  Belgian Restaurant  
1         Snack Place  
2          Restaurant  
3    Greek Restaurant  
4                Park

In [67]:
Brussel_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                  
Anderlecht                                16                      16     16   
Auderghem                                 34                      34     34   
Berchem-Sainte-Agathe                     11                      11     11   
Bruxelles                                100                     100    100   
Etterbeek                                 45                      45     45   
Evere                                     27                      27     27   
Forest                                    14                      14     14   
Ganshoren                                 26                      26     26   
Ixelles                                   30                      30     30   
Jette                                     39                      39     39   
Koekelberg                                16                      16     16   
Molenbeek-Saint-Jean                       9                       9      9   
Saint-Gilles                              73                      73     73   
Saint-Josse-Ten-Noode                     33                      33     33   
Schaerbeek                                38                      38     38   
Uccle                                     44                      44     44   
Watermael-Boitsfort                       28                      28     28   
Woluwe-Saint-Lambert                      18                      18     18   
Woluwe-Saint-Pierre                        4                       4      4   

                       Venue Latitude  Venue Longitude  Venue Category  
Neighborhood                                                            
Anderlecht                         16               16              16  
Auderghem                          34               34              34  
Berchem-Sainte-Agathe              11               11              11  
Bruxelles                         100              100             100  
Etterbeek                          45               45              45  
Evere                              27               27              27  
Forest                             14               14              14  
Ganshoren                          26               26              26  
Ixelles                            30               30              30  
Jette                              39               39              39  
Koekelberg                         16               16              16  
Molenbeek-Saint-Jean                9                9               9  
Saint-Gilles                       73               73              73  
Saint-Josse-Ten-Noode              33               33              33  
Schaerbeek                         38               38              38  
Uccle                              44               44              44  
Watermael-Boitsfort                28               28              28  
Woluwe-Saint-Lambert               18               18              18  
Woluwe-Saint-Pierre                 4                4               4

#### Let's find out how many unique categories can be curated from all the returned venues

In [68]:
print('There are {} uniques categories.'.format(len(Brussel_venues['Venue Category'].unique())))

There are 163 uniques categories.


#### analyze each neighborhood

In [69]:
# one hot encoding
Brussel_onehot = pd.get_dummies(Brussel_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
Brussel_onehot['Neighborhood'] = Brussel_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [Brussel_onehot.columns[-1]] + list(Brussel_onehot.columns[:-1])
Brussel_onehot = Brussel_onehot[fixed_columns]

Brussel_onehot.head()

Neighborhood  African Restaurant  American Restaurant  Art Gallery  \
0   Anderlecht                   0                    0            0   
1   Anderlecht                   0                    0            0   
2   Anderlecht                   0                    0            0   
3   Anderlecht                   0                    0            0   
4   Anderlecht                   0                    0            0   

   Art Museum  Asian Restaurant  Athletics & Sports  BBQ Joint  Bagel Shop  \
0           0                 0                   0          0           0   
1           0                 0                   0          0           0   
2           0                 0                   0          0           0   
3           0                 0                   0          0           0   
4           0                 0                   0          0           0   

   Bakery  ...  Train Station  Tram Station  Turkish Restaurant  \
0       0  ...              0             0                   0   
1       0  ...              0             0                   0   
2       0  ...              0             0                   0   
3       0  ...              0             0                   0   
4       0  ...              0             0                   0   

   Vegetarian / Vegan Restaurant  Video Game Store  Video Store  \
0                              0                 0            0   
1                              0                 0            0   
2                              0                 0            0   
3                              0                 0            0   
4                              0                 0            0   

   Vietnamese Restaurant  Volleyball Court  Wine Bar  Winery  
0                      0                 0         0       0  
1                      0                 0         0       0  
2                      0                 0         0       0  
3                      0                 0         0       0  
4                      0                 0         0       0  

[5 rows x 164 columns]

In [70]:
Brussel_onehot.shape

(605, 164)

#### Next, let's group rows by neighborhood and by taking the mean of the frequency of occurrence of each category

In [71]:
Brussel_grouped = Brussel_onehot.groupby('Neighborhood').mean().reset_index()
Brussel_grouped

Neighborhood  African Restaurant  American Restaurant  \
0              Anderlecht            0.000000                 0.00   
1               Auderghem            0.000000                 0.00   
2   Berchem-Sainte-Agathe            0.000000                 0.00   
3               Bruxelles            0.000000                 0.01   
4               Etterbeek            0.000000                 0.00   
5                   Evere            0.000000                 0.00   
6                  Forest            0.000000                 0.00   
7               Ganshoren            0.000000                 0.00   
8                 Ixelles            0.000000                 0.00   
9                   Jette            0.000000                 0.00   
10             Koekelberg            0.000000                 0.00   
11   Molenbeek-Saint-Jean            0.000000                 0.00   
12           Saint-Gilles            0.013699                 0.00   
13  Saint-Josse-Ten-Noode            0.000000                 0.00   
14             Schaerbeek            0.000000                 0.00   
15                  Uccle            0.000000                 0.00   
16    Watermael-Boitsfort            0.000000                 0.00   
17   Woluwe-Saint-Lambert            0.000000                 0.00   
18    Woluwe-Saint-Pierre            0.000000                 0.00   

    Art Gallery  Art Museum  Asian Restaurant  Athletics & Sports  BBQ Joint  \
0      0.000000    0.000000          0.000000            0.000000   0.000000   
1      0.000000    0.000000          0.000000            0.029412   0.000000   
2      0.000000    0.000000          0.000000            0.000000   0.000000   
3      0.000000    0.000000          0.020000            0.000000   0.000000   
4      0.022222    0.022222          0.022222            0.000000   0.000000   
5      0.000000    0.000000          0.000000            0.000000   0.000000   
6      0.000000    0.000000          0.000000            0.071429   0.000000   
7      0.000000    0.000000          0.000000            0.000000   0.000000   
8      0.000000    0.000000          0.000000            0.000000   0.000000   
9      0.025641    0.000000          0.000000            0.000000   0.025641   
10     0.000000    0.000000          0.000000            0.000000   0.000000   
11     0.000000    0.000000          0.000000            0.000000   0.000000   
12     0.013699    0.000000          0.000000            0.000000   0.013699   
13     0.000000    0.000000          0.000000            0.000000   0.000000   
14     0.000000    0.000000          0.026316            0.000000   0.000000   
15     0.000000    0.000000          0.000000            0.000000   0.000000   
16     0.000000    0.000000          0.000000            0.000000   0.000000   
17     0.000000    0.000000          0.000000            0.055556   0.000000   
18     0.000000    0.000000          0.000000            0.000000   0.000000   

    Bagel Shop    Bakery  ...  Train Station  Tram Station  \
0     0.000000  0.000000  ...       0.000000      0.000000   
1     0.000000  0.029412  ...       0.000000      0.029412   
2     0.000000  0.000000  ...       0.000000      0.090909   
3     0.000000  0.020000  ...       0.000000      0.000000   
4     0.000000  0.022222  ...       0.000000      0.000000   
5     0.000000  0.074074  ...       0.000000      0.037037   
6     0.000000  0.000000  ...       0.000000      0.000000   
7     0.000000  0.038462  ...       0.000000      0.000000   
8     0.000000  0.000000  ...       0.000000      0.033333   
9     0.000000  0.025641  ...       0.025641      0.102564   
10    0.000000  0.000000  ...       0.000000      0.000000   
11    0.000000  0.000000  ...       0.000000      0.000000   
12    0.013699  0.027397  ...       0.000000      0.027397   
13    0.000000  0.030303  ...       0.000000      0.000000   
14    0.000000  0.078947  ...       0.000000      0.078947   
15    0.022727  0.0

In [72]:
Brussel_grouped.shape

(19, 164)

#### Let's create new data frame and display top 10 venues for each neighborhood

In [73]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [74]:
import numpy as np
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = Brussel_grouped['Neighborhood']

for ind in np.arange(Brussel_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(Brussel_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0             Anderlecht         Grocery Store      Greek Restaurant   
1              Auderghem     French Restaurant         Grocery Store   
2  Berchem-Sainte-Agathe      Greek Restaurant          Burger Joint   
3              Bruxelles        Chocolate Shop                   Bar   
4              Etterbeek    Italian Restaurant                 Plaza   

       3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0          Convenience Store            Restaurant        Cosmetics Shop   
1  Middle Eastern Restaurant      Sushi Restaurant    Belgian Restaurant   
2                Snack Place                   Gym                 Plaza   
3                   Beer Bar        Sandwich Place    Belgian Restaurant   
4                Pizza Place                   Bar                Lounge   

       6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0  Middle Eastern Restaurant           Snack Place        Discount Store   
1         Italian Restaurant    Athletics & Sports        Ice Cream Shop   
2               Tram Station           Supermarket     French Restaurant   
3                  Gastropub                 Plaza      Greek Restaurant   
4           Greek Restaurant     French Restaurant              Bus Stop   

  9th Most Common Venue  10th Most Common Venue  
0    Belgian Restaurant          Sandwich Place  
1             Nightclub         Cultural Center  
2            Restaurant  Furniture / Home Store  
3             Bookstore         Thai Restaurant  
4        Sandwich Place             Snack Place

#### Cluster Neighborhoods - Run k-means to cluster the neighborhoods into 7 clusters

In [75]:
# set number of clusters
kclusters = 7

Brussel_grouped_clustering = Brussel_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(Brussel_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([0, 0, 4, 2, 2, 2, 5, 2, 2, 2], dtype=int32)

Let's create a new dataframe that includes the cluster as well as the top 10 venues for each neighborhood.

In [76]:
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)
Brussel_merged = Brussel_df

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
Brussel_merged = Brussel_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood', how='right')

Brussel_merged.head() # check the last columns!

Postal code           Neighborhood   Latitude  Longitude  Cluster Labels  \
0         1070             Anderlecht  50.838141   4.312340               0   
1         1160              Auderghem  50.815657   4.433139               0   
2         1082  Berchem-Sainte-Agathe  50.863984   4.292702               4   
3         1000              Bruxelles  50.846557   4.351697               2   
4         1040              Etterbeek  50.836851   4.389510               2   

  1st Most Common Venue 2nd Most Common Venue      3rd Most Common Venue  \
0         Grocery Store      Greek Restaurant          Convenience Store   
1     French Restaurant         Grocery Store  Middle Eastern Restaurant   
2      Greek Restaurant          Burger Joint                Snack Place   
3        Chocolate Shop                   Bar                   Beer Bar   
4    Italian Restaurant                 Plaza                Pizza Place   

  4th Most Common Venue 5th Most Common Venue      6th Most Common Venue  \
0            Restaurant        Cosmetics Shop  Middle Eastern Restaurant   
1      Sushi Restaurant    Belgian Restaurant         Italian Restaurant   
2                   Gym                 Plaza               Tram Station   
3        Sandwich Place    Belgian Restaurant                  Gastropub   
4                   Bar                Lounge           Greek Restaurant   

  7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
0           Snack Place        Discount Store    Belgian Restaurant   
1    Athletics & Sports        Ice Cream Shop             Nightclub   
2           Supermarket     French Restaurant            Restaurant   
3                 Plaza      Greek Restaurant             Bookstore   
4     French Restaurant              Bus Stop        Sandwich Place   

   10th Most Common Venue  
0          Sandwich Place  
1         Cultural Center  
2  Furniture / Home Store  
3         Thai Restaurant  
4             Snack Place

#### Let's visualize resulting clusters

In [77]:
# create map
map_clusters = folium.Map(location=[latitude_Brussel, longitude_Brussel], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i+x+(i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(Brussel_merged['Latitude'], Brussel_merged['Longitude'], Brussel_merged['Neighborhood'], Brussel_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

In [78]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 0, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
0          1070         Grocery Store      Greek Restaurant   
1          1160     French Restaurant         Grocery Store   
17         1200                  Park     French Restaurant   

        3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0           Convenience Store            Restaurant        Cosmetics Shop   
1   Middle Eastern Restaurant      Sushi Restaurant    Belgian Restaurant   
17                Supermarket    Italian Restaurant            Restaurant   

        6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0   Middle Eastern Restaurant           Snack Place        Discount Store   
1          Italian Restaurant    Athletics & Sports        Ice Cream Shop   
17                Bus Station    Chinese Restaurant      Basketball Court   

   9th Most Common Venue 10th Most Common Venue  
0     Belgian Restaurant         Sandwich Place  
1              Nightclub        Cultural Center  
17          Climbing Gym    Fried Chicken Joint

In [79]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 1, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
11         1080           Supermarket                   Spa   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
11          Dance Studio   Fried Chicken Joint           Snack Place   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
11                Winery     Electronics Store    Falafel Restaurant   

   9th Most Common Venue 10th Most Common Venue  
11               Factory                Exhibit

In [80]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 2, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
3          1000        Chocolate Shop                   Bar   
4          1040    Italian Restaurant                 Plaza   
5          1140            Restaurant                Bakery   
7          1083    Italian Restaurant                   Bar   
8          1050                 Plaza                   Bar   
9          1090          Tram Station                   Bar   
12         1060                   Bar             Brasserie   
13         1210        Sandwich Place    Italian Restaurant   
14         1030           Supermarket                 Plaza   
15         1180                 Plaza        Cosmetics Shop   
16         1170    Italian Restaurant            Restaurant   

   3rd Most Common Venue  4th Most Common Venue 5th Most Common Venue  \
3               Beer Bar         Sandwich Place    Belgian Restaurant   
4            Pizza Place                    Bar                Lounge   
5           Hockey Field                   Pool            Sports Bar   
7            Bus Station               Bus Stop                  Café   
8      French Restaurant  Vietnamese Restaurant       Thai Restaurant   
9                    Gym            Pizza Place           Snack Place   
12                 Plaza            Pizza Place    Italian Restaurant   
13      Kebab Restaurant             Restaurant           Supermarket   
14                Bakery           Tram Station             Gastropub   
15     French Restaurant            Supermarket        Sandwich Place   
16    Chinese Restaurant                   Park       Thai Restaurant   

   6th Most Common Venue          7th Most Common Venue  \
3              Gastropub                          Plaza   
4       Greek Restaurant              French Restaurant   
5              Brasserie                    Snack Place   
7         Sandwich Place             Chinese Restaurant   
8       Greek Restaurant  Paper / Office Supplies Store   
9            Bus Station                    Supermarket   
12     French Restaurant                 Sandwich Place   
13           Pizza Place                          Plaza   
14    Italian Restaurant      Middle Eastern Restaurant   
15    Italian Restaurant                         Bakery   
16              Bus Stop                   Soccer Field   

        8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
3            Greek Restaurant             Bookstore        Thai Restaurant  
4                    Bus Stop        Sandwich Place            Snack Place  
5                 Bus Station        Sandwich Place                  Plaza  
7                 Flower Shop            Steakhouse            Supermarket  
8               Grocery Store           Pizza Place             Playground  
9                   Gastropub         Grocery Store   Gym / Fitness Center  
12        Moroccan Restaurant  Gym / Fitness Center          Grocery Store  
13  Middle Eastern Restaurant           Snack Place    Lebanese Restaurant  
14          Electronics Store           Coffee Shop      French Restaurant  
15          Food & Drink Shop        Chocolate Shop             Kids Store  
16             Farmers Market     Food & Drink Shop              Bookstore

In [81]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 3, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
18         1150                  Park          Tram Station   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
18                 Trail           Sports Club                Winery   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
18        Discount Store        Farmers Market    Falafel Restaurant   

   9th Most Common Venue 10th Most Common Venue  
18               Factory                Exhibit

In [82]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 4, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
2         1082      Greek Restaurant          Burger Joint   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
2           Snack Place                   Gym                 Plaza   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
2          Tram Station           Supermarket     French Restaurant   

  9th Most Common Venue  10th Most Common Venue  
2            Restaurant  Furniture / Home Store

In [83]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 5, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
6         1190           Supermarket     French Restaurant   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
6      Halal Restaurant                  Park     Health Food Store   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
6               Factory    Athletics & Sports                 Plaza   

  9th Most Common Venue 10th Most Common Venue  
6           Men's Store              Cafeteria

In [84]:
Brussel_merged.loc[Brussel_merged['Cluster Labels'] == 6, Brussel_merged.columns[[0] + list(range(5, Brussel_merged.shape[1]))]]

Postal code 1st Most Common Venue 2nd Most Common Venue  \
10         1081  Gym / Fitness Center                   Bar   

   3rd Most Common Venue 4th Most Common Venue  5th Most Common Venue  \
10     Convenience Store                  Park  Indonesian Restaurant   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
10       Bed & Breakfast    Falafel Restaurant        Sandwich Place   

   9th Most Common Venue 10th Most Common Venue  
10     French Restaurant                 Hostel

### Results

The following are the highlights of the 7 clusters above:
1. The most common venues are clearly located only in 3rd Cluster (center of Brussels), which makes the choice of the final location very easy.
2. As for restaurants, bars and coffee shops are very popular also in 3rd Cluster (center of Brussels), Especially in 1000 Brussels, 1040 Etterbeek and 1050 Ixelles.
3. Although, the Clusters have variations, a very visible presence is the predominance of bars and restaurants, so new hotel can be opened in those places.

### Discussion and Conclusion

It is noticable that 3rd Cluster is the most viable clusters to build a new luxury hotel with guarantees. The proximity to a big number of Restaurants (lunch and dinner venues for guests), Coffee shops and other amenities and accessibility to station are also very important points to take into account when making the right choice. 

The municipalities like 1000 Brussels, 1040 Etterbeek and 1050 Ixelles, lies in the center of Brussels and have proximity to all kind of most common venues visited by locals as well as tourists. These neighborhoods could be the best places to open new luxury hotel in the city. 

In conclusion, this project would have had better results if there were more available data in terms of actual land pricing data within the area, public transportantion access and allowance of more venues exploration with the Foursquare (limited venues for free calls).
However, based on the available data, my advice to Marriott group would be to focus on 1000 Brussels, 1040 Etterbeek and 1050 Ixelles when investing on a new luxury hotel.